ios/Sources/Services/Network/HTTPClient.swift

```
@@ -0,0 +1,59 @@
+//
+//  HTTPClient.swift
+//  YourFirstCommit
+//
+//  Created by Alex Antipov on 15.02.2023.
+//
+
+import Foundation
+
+protocol HTTPClient {
+    func sendRequest<T: Decodable>(endpoint: Endpoint, responseModel: T.Type) async -> Result<T, RequestError>
```

we can use `async throws` instead of return an error within the Result type